plugins/inputs/cisco_telemetry_mdt_netconf/cisco_telemetry_mdt_netconf.go

```diff
@@ -0,0 +1,930 @@
+
+					if err != nil {
+						c.reportError(ctx, err, "create telemetry subscription")
+					} else {
+						c.Log.Debugf(
+							"Established subscription",
+						)
+						// Announce state:created to the next stage of the pipeline
+						dsrs.setting.state = created
+					}
+				} else {
+					err = fmt.Errorf(
+						"failed to create telemetry subscription: "+
+							"time period for subscription %d has to be positive but is %d",
+						i+1,
+						time.Duration(subscription.Period),
+					)
+					c.acc.AddError(err)
+				}
+			} else {
+				err = fmt.Errorf(
+					"failed to create telemetry subscription: "+
+						"missing field in subscription %d: period",
+					i+1,
+				)
+				c.acc.AddError(err)
+			}
+		}
+	}
+
+	if len(dsrs.Notifications) > 0 {
+		// Initialization of notifications
+		notifications := make([]*netconf.CreateSubscription, len(dsrs.Notifications))
+
+		// Create notification subscriptions over NETCONF
+		for i, notification := range dsrs.Notifications {
+			if notification.Stream != "" {
+				notifications[i] = &netconf.CreateSubscription{
+					Stream: &notification.Stream,
+				}
+			} else {
+				// Allow subscription without any stream, as per specification
+				notifications[i] = &netconf.CreateSubscription{}
+			}
+
+			c.Log.Debugf(
+				"Establishing notification subscription %s...",
+				notification,
+			)
+
+			dsrs.Mutex.Lock()
+			err = dsrs.Session.CallSimple(notifications[i])
+			dsrs.Mutex.Unlock()
+
+			if err != nil {
+				c.reportError(ctx, err, "create notification subscription")
+			} else {
+				c.Log.Debugf(
+					"Established subscription",
+				)
+				// Announce state:created to the next stage of the pipeline
+				dsrs.setting.state = created
+			}
+		}
+	}
+
+	if len(dsrs.Subscriptions) == 0 && len(dsrs.Notifications) == 0 {
+		err = fmt.Errorf("missing subscription requests")
+		c.reportError(ctx, err, "create subscription")
+	}
+
+	return err
+}
+
+// receiveTelemetry for subscriptions as yang-push notifications over NETCONF
+func (dsrs *dialinSubscriptionRequestsService) receiveTelemetry(
+	ctx context.Context, c *CiscoTelemetryNETCONF) {
+	push := &netconfYangPush{}
+
+	// Receive telemetry data
+	for ctx.Err() == nil {
+		dsrs.Mutex.Lock()
+		sessionExists := dsrs.Session != nil
+		dsrs.Mutex.Unlock()
+
+		if sessionExists {
+			c.Log.Debugf(
+				"Receiving yang-push messages ...",
+			)
+
+			err := dsrs.Session.Receive(push)
+
+			if err != nil {
+				c.reportError(ctx, err, "receive telemetry subscription data")
+			} else {
+				// Unmarshal XML data
+				var tt TelemetryTree
+
+				err = xml.Unmarshal(push.PushUpdate.Content.InnerXML, &tt)
+				if err != nil && err != io.EOF {
+					// Don't throw an error if error is EOF due to empty xml
+					c.acc.AddError(fmt.Errorf(
+						"failed to unmarshal XML: %s",
+						err,
+					))
+				} else if err == nil {
+					// Send data to Influx accumulator
+					c.handleTelemetry(tt, push.EventTime)
+				}
+			}
+		}
```

Comment:
I think the `session` comparison is atomic so you can do the following to also save some indentation and improve readability
```suggestion
		if dsrs.Session == nil {
			continue
		}
		c.Log.Debugf("Receiving yang-push messages ...")

		if err := dsrs.Session.Receive(push); err != nil {
			c.reportError(ctx, err, "receive telemetry subscription data")
			continue
		}
		
		// Unmarshal XML data
		var tt TelemetryTree
		err := xml.Unmarshal(push.PushUpdate.Content.InnerXML, &tt)
		if err != nil && errors.Is(err, io.EOF) {
			// Don't throw an error if error is EOF due to empty xml
			c.acc.AddError(fmt.Errorf("failed to unmarshal XML: %w", err))
			continue
		}
		
		// Send data to Influx accumulator
		c.handleTelemetry(tt, push.EventTime)
```